guardian/transcription-service
# Format and lint with precommit hook

## What does this change?
- Adds a [pre-commit hook](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) to format staged files with prettier and lint them with eslint
  - installs [`husky`](https://typicode.github.io/husky/#/), a tool to configure git hooks
  - installs [`lint-staged`](https://github.com/okonet/lint-staged), [recommended by prettier](https://prettier.io/docs/en/precommit.html) for projects that also want to run eslint in a hook and support partially staged files.

## Why?
This should help us avoid noisy git history due to formatting and linting changes unrelated to the main motivation of a commit. 
Automatic formatting should allow contributors and reviewers to ignore formatting and focus on more substantial code changes.

## How to test
- try to commit changes that break eslint rules. Fixable changes like incorrect import order should be fixed automatically when they're committed. Unfixable ones like @typescript-eslint/no-explicit-any will stop the commit.
